billing-client/event.go

```diff
@@ -0,0 +1,34 @@
+package client
+
+import (
+	"time"
+)
+
+type Event struct {
+	UniqueKey          string            `json:"unique_key" msg:"unique_key"`
+	InternalInstanceID string            `json:"internal_instance_id" msg:"internal_instance_id"`
+	OccurredAt         time.Time         `json:"occurred_at" msg:"occurred_at"`
+	Amounts            Amounts           `json:"amounts" msg:"amounts"`
+	Metadata           map[string]string `json:"metadata" msg:"metadata"`
+}
+
+func (e Event) toRecords() []map[string]interface{} {
+	var records []map[string]interface{}
+	for t, v := range e.Amounts {
+		records = append(records, map[string]interface{}{
+			"unique_key":           e.UniqueKey + ":" + t,
+			"internal_instance_id": e.InternalInstanceID,
+			"amount_type":          t,
+			"amount_value":         v,
+			"occurred_at":          e.OccurredAt,
+			"metadata":             e.Metadata,
+		})
```

Comment:
I'd be more comfortable is this was a struct, although I realise that doesn't really offer much protection...